include/rmm/rmm.hpp

```diff
@@ -205,10 +199,12 @@ inline rmmError_t realloc(T** ptr, size_t new_size, cudaStream_t stream,
 inline rmmError_t free(void* ptr, cudaStream_t stream, const char* file,
                    unsigned int line) {
   rmm::LogIt log(rmm::Logger::Free, ptr, 0, stream, file, line);
-  if (rmm::Manager::usePoolAllocator())
-    RMM_CHECK_CNMEM(cnmemFree(ptr, stream));
-  else
-    RMM_CHECK_CUDA(cudaFree(ptr));
+  if(ptr != nullptr){
```

Comment:
Ok I will make  a unit test that makes sure this is not a problem because I remember adding it to avoid an issue but it may have been one in the custom memory resource I made outside of RMM.

Reply:
Any resource should be able to handle a `nullptr`. All of the default resources are explicitly tested with deallocating a `nullptr`.